src/Raven.Server/Documents/Queries/CollectionQueryEnumerable.cs

```diff
@@ -223,15 +220,30 @@ private class Enumerator : IEnumerator<Document>
             public bool MoveNext()
             {
                 if (_initialized == false)
-                    _start = Initialize();
+                {
+                    Initialize(out var processedAllDocuments);
+                    
+                    // When our query skipped all documents from underlying stream
+                    if (processedAllDocuments)
+                        return false;
```

Comment:
We want to avoid method calls here since it's hot-path.

Reply:
Since it's only one call per enumerator instance, I think it's better to return the value explicitly with the name. Returning it as a return parameter from `Initialize` can be confusing.

Reply:
ok